src/utils.ts

```diff
@@ -59,31 +59,43 @@ export async function inputMapExists(flags: Flags) {
   }
 }
 
+function addEnv(env: string[], newEnv: string, removedEnv: string) {
+  let removedEnvIndex = -1
+  if ((removedEnvIndex = env.indexOf(removedEnv)) !== -1)
+    env.splice(removedEnvIndex, 1)
+
+  env.push(newEnv)
+}
```

Comment:
Not all environments have mutually exclusive "removedEnv" cases.

I think we need to use a more direct exclusion mechanism, where negations by default do not participate in exclusivity.

```suggestion
const exclusions: {
  'production': ['development'],
  'development': ['production'],
  'node': ['browser', 'deno'],
  'deno': ['node', 'browser'],
  'browser': ['node', 'deno'],
};
function removeEnv(env: string[], removeEnv: string) {
  if (env.includes(exclude))
    env.splice(env.indexOf(exclude), 1);
  return env.sort();
}
function addEnv(env: string[], newEnvs: string[]) {
  if (env.indexOf(newEnv) === -1)
    env.push(newEnv);
  const excludes = exclusions[newEnv];
  if (excludes) {
    for (const exclude of excludes) {
      if (env.includes(exclude) && !newEnvs.includes(exclude))
        env.splice(env.indexOf(exclude), 1);
    }
  }
  return env.sort();
}
```